test/suites/dev/moonbase/test-xcm-v4/test-xcm-evm-fee-with-origin-token.ts

```diff
@@ -0,0 +1,207 @@
+        initialSenderBalance,
+        descendAddress
+      );
+
+      // Register foreign asset used to transfer (i.e. xcMatic)
+      xcMaticAssetId = await registerAndFundAsset(
+        context,
+        xcMaticUnitsPerSecond,
+        PARA_1001_SOURCE_LOCATION,
+        initialSenderBalance,
+        descendAddress
+      );
+    });
+
+    it({
+      id: "T01",
+      title: "should receive foreign asset transfer, paying fees in origin chain's foreign asset",
+      test: async function () {
+        // 3. Build incoming XCM message
+        const xcmMessage = new XcmFragment({
+          assets: [
+            {
+              multilocation: PARA_1000_SOURCE_LOCATION.Xcm,
+              fungible: 1_000_000_000n,
+            },
+            {
+              multilocation: PARA_1001_SOURCE_LOCATION.Xcm,
+              fungible: xcMaticToSend,
+            },
+          ],
+          weight_limit: {
+            refTime: 50_000_000_000n,
+            proofSize: 150000n,
+          },
+          descend_origin: sendingAddress,
+        })
+          .descend_origin()
+          .withdraw_asset()
+          .buy_execution()
+          .deposit_asset_definite(
+            PARA_1001_SOURCE_LOCATION.Xcm,
+            xcMaticToSend,
+            ethan.address as `0x${string}`
+          )
+          .deposit_asset_definite(
+            PARA_1000_SOURCE_LOCATION.Xcm,
+            500_000_000n,
+            descendAddress as `0x${string}`
+          )
+          .as_v4();
+
+        await verifyLatestBlockFees(context);
+
+        const xcIntrBalanceBefore = (await api.query.assets.account(xcIntrAssetId, descendAddress))
+          .unwrap()
+          .balance.toBigInt();
+
+        const xcMaticBalanceBefore = (
+          await api.query.assets.account(xcMaticAssetId, descendAddress)
+        )
+          .unwrap()
+          .balance.toBigInt();
+
+        // Simulate reception of an incoming XCM message and create block to execute it
+        await injectHrmpMessageAndSeal(context, 1000, {
+          type: "XcmVersionedXcm",
+          payload: xcmMessage,
+        } as RawXcmMessage);
+
+        const xcIntrBalanceAfter = (await api.query.assets.account(xcIntrAssetId, descendAddress))
+          .unwrap()
+          .balance.toBigInt();
+
+        const xcMaticBalanceAfter = (await api.query.assets.account(xcMaticAssetId, descendAddress))
+          .unwrap()
+          .balance.toBigInt();
+
+        const xcMaticReceivedEthan = (await api.query.assets.account(xcMaticAssetId, ethan.address))
+          .unwrap()
+          .balance.toBigInt();
+
+        // Check that xcIntr where debited from Alith to pay the fees of the XCM execution
+        expect(xcMaticBalanceBefore - xcMaticBalanceAfter).to.be.eq(xcMaticToSend);
+        expect(xcMaticReceivedEthan).to.be.eq(xcMaticToSend);
+        expect(xcIntrBalanceBefore - xcIntrBalanceAfter).to.be.eq(500_000_000n);
+      },
+    });
+  },
+});
+
+async function registerAndFundAsset(
```

Comment:
I think we should move this function to `helpers` folder. I would also suggest renaming it to `registerAndFundOldAsset`.

Also, it would be nice to add a small comment explaining why the function was created.

Reply:
I've updated the test to use the new methods for handling foreign assets, so this is deprecated now.